### ISSUE TITLE
Improve the right-click menu of Notebook cell editor

### ISSUE THREAD

**Issue by karlaspuldaro** [jupyterlab/jupyterlab#8617]
### Problem
As a follow up to issue #2778, and as suggested by @echarles [in this PR comment](https://github.com/jupyterlab/jupyterlab/pull/8425#issuecomment-646950218), it would be helpful to add basic functions to the text content of a notebook cell, such as **cut/copy/paste, select all, undo/redo** , in addition to the existing full cell operations. 
The current cell right click-menu currently looks like:
![image](https://user-images.githubusercontent.com/25207344/85801007-06394e00-b710-11ea-8697-1193ce3e2347.png)

### Proposed Solution
Add the proposed commands to the notebook cell right-click menu as a text editor group of commands, which would be applied to the content of the cell in focus.
![image](https://user-images.githubusercontent.com/25207344/85800343-c58d0500-b70e-11ea-8c55-66d9b218910f.png)

**Comment by ianhi** [comment]
As mentioned in https://github.com/jupyterlab/jupyterlab/pull/8425#issuecomment-650233072 Paste via `window.navigator.clipboard.readText` will not work on Firefox. Maybe a minimal approach to dealing with that would be to disable the paste functionality on firefox?

**Comment by blink1073** [comment]
There is a workaround used in notebook classic, I'll follow up with a link

**Comment by blink1073** [comment]
Here it is: https://github.com/jupyter/notebook/blob/70d74d21ac051fbeaa81d4ef4fff9fa759de96d7/notebook/static/notebook/js/clipboard.js#L98

**Comment by blink1073** [comment]
We could make this a part of https://github.com/jupyterlab/jupyterlab/blob/c9099accef55f19e5cb312bc7e9e8c9762305607/packages/apputils/src/clipboard.ts#L12

**Comment by ianhi** [comment]
Another thought: Markdown cells allow for embedding of images by pasting when a paste has content with mimetype `image/*` which I don't think that `readText` gives access to. My worry is that if there is a paste option accessible via the lumino contextmenu which doesn't support pasting images it will prevent discovery of that feature. Maybe the menu could have some extra context in the name. Like "Paste (text only)"?

Relevant image attachment code is here:
https://github.com/jupyterlab/jupyterlab/blob/c9099accef55f19e5cb312bc7e9e8c9762305607/packages/cells/src/widget.ts#L1225
and here
https://github.com/jupyterlab/jupyterlab/blob/c9099accef55f19e5cb312bc7e9e8c9762305607/packages/cells/src/widget.ts#L1344-L1346

**Comment by ianhi** [comment]
Or even raise a dialog explaining that you need to use ctrl-V or shift-right click to paste an image. This is basically what google docs does:
![image](https://user-images.githubusercontent.com/10111092/85893238-7dc6b600-b7c0-11ea-9de9-515c0e5869ea.png)

**Comment by karlaspuldaro** [comment]
@ianhi Good point, I agree with the approaches for dealing with pasting images, it'd be helpful to provide some kind of feedback/guidance to the user
